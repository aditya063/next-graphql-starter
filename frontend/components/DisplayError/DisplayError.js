/* eslint-disable no-console */
/* eslint-disable react/no-array-index-key */
/* eslint-disable react/no-unused-prop-types */
/* eslint-disable react/forbid-prop-types */
import PropTypes from 'prop-types';
import * as Styled from './DisplayError.style';

const DisplayError = React.memo(error => {
  if (error.error.graphQLErrors) {
    return (
      <Styled.divError>
        {error.error.graphQLErrors.map((e, i) => (
          <p key={i}>{e.message}</p>
        ))}
      </Styled.divError>
    );
  }

  if (error.error)
    return (
      <Styled.divError>
        <p>{error.error.message}</p>
      </Styled.divError>
    );

  return (
    <>
      {console.log('Opps: ', error)}
      <Styled.divError>Opps, something went wrong.</Styled.divError>
    </>
  );
});

DisplayError.defaultProps = {
  error: {}
};

DisplayError.propTypes = {
  error: PropTypes.object
};

export default DisplayError;
