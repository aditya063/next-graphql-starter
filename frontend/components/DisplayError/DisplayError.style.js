import styled from 'styled-components';

export const divError = styled.div`
  padding-left: 1rem;
  margin: 2rem 0;
  border-left: 5px solid red;
`;
