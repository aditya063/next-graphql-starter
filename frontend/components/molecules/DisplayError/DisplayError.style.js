import styled from 'styled-components';

export const DisplayError = styled.ul`
  padding: 0;
  list-style: none;
`;
