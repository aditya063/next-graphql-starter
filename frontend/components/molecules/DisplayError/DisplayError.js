import PropTypes from 'prop-types';
import AlertMsg from '../../atoms/AlertMsg/AlertMsg';
import * as sc from './DisplayError.style';

const DisplayError = props => {
  if (props.error.graphQLErrors && props.error.graphQLErrors.length) {
    return (
      <sc.DisplayError data-testid='DisplayError'>
        {props.error.graphQLErrors.map((error, i) => (
          <AlertMsg key={`error${i}`} message={error.message} type='error' />
        ))}
      </sc.DisplayError>
    );
  }

  if (props.error.message) {
    return (
      <sc.DisplayError data-testid='DisplayError'>
        <AlertMsg message={props.error.message} type='error' />
      </sc.DisplayError>
    );
  }

  return (
    <sc.DisplayError data-testid='DisplayError'>
      <AlertMsg message='Opps, something went wrong.' type='error' />
    </sc.DisplayError>
  );
};

DisplayError.defaultProps = { error: {} };

DisplayError.propTypes = {
  error: PropTypes.shape({
    graphQLErrors: PropTypes.array,
    message: PropTypes.string
  })
};

export default React.memo(DisplayError);
