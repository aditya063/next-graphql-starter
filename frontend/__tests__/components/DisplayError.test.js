import { prettyDOM, render, fireEvent, wait } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import DisplayError from '../../components/DisplayError/DisplayError';
import theme from '../../public/styles/theme.style';

describe('DisplayError', () => {
  it('renders generic error if no error message prop', () => {
    const errorMessage = 'Opps, something went wrong.';
    const mockProps = {};

    const utils = render(
      <ThemeProvider theme={theme}>
        <DisplayError {...mockProps} />
      </ThemeProvider>
    );

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent(errorMessage);
  });

  it('renders general errors', () => {
    const errorMessage = 'mock general error';
    const mockProps = { error: { message: 'mock general error' } };

    const utils = render(
      <ThemeProvider theme={theme}>
        <DisplayError {...mockProps} />
      </ThemeProvider>
    );

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent(errorMessage);
  });

  it('renders graphQLErrors', () => {
    const errorMessage = 'mock graphQLError';
    const mockProps = {
      error: { graphQLErrors: [{ message: errorMessage }] }
    };

    const utils = render(
      <ThemeProvider theme={theme}>
        <DisplayError {...mockProps} />
      </ThemeProvider>
    );

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent(errorMessage);
  });
});
