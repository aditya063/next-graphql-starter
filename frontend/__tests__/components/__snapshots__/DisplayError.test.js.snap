// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayError renders general errors 1`] = `
.c0 {
  border-left: 5px solid red;
  list-style: none;
  padding-left: 1rem;
}

<ul
  class="c0"
>
  <li>
    mock general error
  </li>
</ul>
`;

exports[`DisplayError renders generic error if no error message prop 1`] = `
.c0 {
  border-left: 5px solid red;
  list-style: none;
  padding-left: 1rem;
}

<ul
  class="c0"
>
  <li>
    Opps, something went wrong.
  </li>
</ul>
`;

exports[`DisplayError renders graphQLErrors 1`] = `
.c0 {
  border-left: 5px solid red;
  list-style: none;
  padding-left: 1rem;
}

<ul
  class="c0"
>
  <li>
    mock graphQLError
  </li>
</ul>
`;
